db/dcrsqlite/apisource.go

```diff
@@ -1309,42 +1309,60 @@ func (db *wiredDB) GetExplorerTx(txid string) *explorer.TxInfo {
 
 func (db *wiredDB) GetExplorerAddress(address string, count, offset int64) (*explorer.AddressInfo, error) {
 	addr, err := dcrutil.DecodeAddress(address)
+
+	// checks for P2Pk type addresses
+	if strings.HasPrefix(address, "Dk") || strings.HasPrefix(address, "Tk") || strings.HasPrefix(address, "Sk") {
+		return nil, fmt.Errorf("P2PK address detected")
+	}
 	if err != nil {
+		log.Infof("Invalid address %s: %v", address, err)
+
+		// This is here to detect a bitcoin type address
+		if (strings.HasPrefix(address, "bc") || strings.HasPrefix(address, "1") || strings.HasPrefix(address, "3")) && len(address) >= 25 && len(address) <= 34 {
+			return nil, fmt.Errorf("Possible Bitcoin address detected.")
+		}
 		return nil, err
 	}
 
-	// Short circuit the transaction and balance queries if the provided address
-	// is the zero pubkey hash address commonly used for zero value
-	// sstxchange-tagged outputs.
-	isDummyAddress := IsZeroHashP2PHKAddress(address, db.params)
-	if isDummyAddress {
-		return &explorer.AddressInfo{
-			Address:         address,
-			Balance:         new(explorer.AddressBalance),
-			UnconfirmedTxns: new(explorer.AddressTransactions),
-			IsDummyAddress:  true,
-			Fullmode:        true,
-		}, nil
+	{
+		// Short circuit the transaction and balance queries if the provided address
+		// is the zero pubkey hash address commonly used for zero value
+		// sstxchange-tagged outputs.
+		isDummyAddress := IsZeroHashP2PHKAddress(address, db.params)
+		if isDummyAddress {
+			return &explorer.AddressInfo{
+				Address:         address,
+				Balance:         new(explorer.AddressBalance),
+				UnconfirmedTxns: new(explorer.AddressTransactions),
+				IsDummyAddress:  true,
+				Fullmode:        true,
+			}, nil
+		}
 	}
 
-	// Handle invalid address.
-	if !ValidateNetworkAddress(addr, db.params) {
-		return nil, fmt.Errorf("wrong network: address %s is not valid for %v",
-			address, db.params.Name)
+	// Dectection of when an address belonging to a different network is inputed
+	if addr.Net().Name != db.params.Name {
+		return nil, fmt.Errorf("Wrong Network. You pasted a %s address on %v", addr.Net().Name, db.params.Name)
```

Comment:
`error`s should not start with caps.  Also, I would prefer not to say things like "You pasted".  Firstly, the "You" part should be something less personal, just stating that what condition got us here (address is valid on network X, not network Y).  Also, the "pasted" action is too closely tied to UI, when a program or other package could simply be using `(*wiredDB).GetExplorerAddress)` in whatever way they choose.